src/TurbulenceConvection/Parameters.jl

```diff
@@ -23,6 +23,44 @@ Base.@kwdef struct TurbulenceConvectionParameters{FT, MP, SFP} <: ATCP
     microph_scaling_melt::FT # TODO: move to microphysics parameter set? or Microphysics1Moment?
     microphys_params::MP
     surf_flux_params::SFP
+    case_name::String
```

Comment:
Let's remove the strings since they're GPU-incompatible. We can use types instead, and that way we can also leverage dispatch.

Also, we should use `::Int` over `::Integer` since `Integer` is an abstract type and we should avoid structs with abstract types (see https://docs.julialang.org/en/v1/manual/performance-tips/#Avoid-fields-with-abstract-type)

Reply:
Also, `case` is not needed, nor is it currently used, past initialization, and we don't want simulations to be case-dependent (this is why _all_ case information in SCAMPy was in a `Cases` file). We should avoid making it available in our RHS functions to ensure it doesn't happen.

Reply:
I can turn the String fields into types, and fix the Integer fields.
Should I get rid of the `case_name `and `case ` occurrences entirely then?

Reply:
It looks like it's being deleted in #1327, maybe the simplest thing for now is to grab `case_name` from `parsed_args` (which is available in `turbconv_cache`). Do we need to do anything with `case`?

Reply:
I don't think we do, I couldn't find any examples of it actually being used.